stdlib/public/core/StringUTF8View.swift

```diff
@@ -251,13 +251,16 @@ extension String {
   ///     }
   ///     // Prints "6"
   public var utf8CString: ContiguousArray<CChar> {
-    if _fastPath(_guts.isFastUTF8) {
-      var result = _guts.withFastCChar { ContiguousArray($0) }
-      result.append(0)
-      return result
-    }
+    @_effects(readonly) @_semantics("string.getUTF8CString")
```

Comment:
Is this effect true, given that this allocates a return value and  `_slowUTF8CString` is not effect annotated? CC @aschwaighofer .

Out of curiosity, if this were `@inlinable`, could a more general optimization apply?

Reply:
Yes, readonly is okay, because memory effects only refer to effects, which are observable from "outside". An allocation does not write to any memory which is also observable form "outside".

Reply:
about inlinable: maybe, I didn't try. But it would be unrelated to this optimization